include/wx/graphics.h

```diff
@@ -132,6 +132,49 @@ class WXDLLIMPEXP_CORE wxGraphicsObject : public wxObject
     wxDECLARE_DYNAMIC_CLASS(wxGraphicsObject);
 };
 
+// ----------------------------------------------------------------------------
+// wxGraphicsPenInfo describes a wxGraphicsPen
+// ----------------------------------------------------------------------------
+
+class wxGraphicsPenInfo : public wxPenInfo
+{
+public:
+    wxGraphicsPenInfo()
+    : wxPenInfo()
+    {
+        m_widthF = -1.0;
+    }
+
+    explicit wxGraphicsPenInfo(const wxColour& colour, double widthF = 1.0, wxPenStyle style = wxPENSTYLE_SOLID)
+    : wxPenInfo(colour, 0, style)
+    {
+        m_widthF = widthF;
+    }
+
+    // Setters for the various attributes. All of them return the object itself
+    // so that the calls to them could be chained.
+
+    wxGraphicsPenInfo& Colour(const wxColour& colour);
+
+    wxGraphicsPenInfo& Width(int width);
+
+    wxGraphicsPenInfo& Style(wxPenStyle style);
+    wxGraphicsPenInfo& Stipple(const wxBitmap& stipple);
+    wxGraphicsPenInfo& Dashes(int nb_dashes, const wxDash *dash);
+    wxGraphicsPenInfo& Join(wxPenJoin join);
+    wxGraphicsPenInfo& Cap(wxPenCap cap);
+
+    wxGraphicsPenInfo& WidthF(wxDouble widthF)
```

Comment:
Maybe I could just make width a double in wxGraphicsPenInfo? It seems all underlying implementations just store a double anyway.